firestore/client.go

```diff
@@ -381,3 +399,25 @@ func (ec emulatorCreds) GetRequestMetadata(ctx context.Context, uri ...string) (
 func (ec emulatorCreds) RequireTransportSecurity() bool {
 	return false
 }
+
+// ReadOptions provides specific instructions for how to access documents in the database.
+// Currently, only ReadTime is supported.
+type ReadOptions struct {
+	ReadTime time.Time // ReadTime specifies the time at which to capture a "snapshot" of the documents in the database.
+}
+
+// ReadOptionable interface allows for abstraction of computing read time settings.
+type readOptionable interface {
+	getReadOptions() *ReadOptions
+}
+
+// parseReadTime ensures that fallback order of read options is respected.
+// First, if a ReadOption is set on the readOptionable
+func parseReadTime(c *Client, ro readOptionable) (*timestamppb.Timestamp, bool) {
```

Comment:
Could we just pass two `time.Time` values here? It seems the option is preferred first and then the client is, otherwise no read time is set. Is that correct?

Reply:
Yes, that's correct. The `ReadOption` should be read first and then the Client's ReadOption, (if any). We could pass two `time.Time` objects; semantically I think it makes more sense as written.